JenkinsfileStressTests

```diff
@@ -0,0 +1,83 @@
+def maxRetries = 120
+def waitForCommandSuccess(command, maxRetries) {
+    def tries = 0;
+    while (tries++ < maxRetries) {
+        try {
+            sh command
+            return true
+        } catch (Exception e) {
+            if (tries >= maxRetries) {
+                throw new Exception("${maxRetries} tries exceeded. ${command} failed.", e)
+            }
+            sh "sleep 1"
+        }
+    }
+}
+
+pipeline {
+    agent { label 'my-slave' }
+
+    tools {
+        maven 'maven'
+    }
+
+    stages {
+        stage('Compile') {
+            steps {
+                sh "mvn clean install -DskipTests"
+            }
+        }
+        stage('Build image') {
+            steps {
+                script {
+                    docker.withTool('docker') {
+                        sh "cp server/container/guice/cassandra-rabbitmq-guice/target/james-server-cassandra-rabbitmq-guice.jar dockerfiles/run/guice/cassandra-rabbitmq/destination"
+                        sh "cp -r server/container/guice/cassandra-rabbitmq-guice/target/james-server-cassandra-rabbitmq-guice.lib dockerfiles/run/guice/cassandra-rabbitmq/destination"
+                        sh "cp server/container/cli/target/james-server-cli.jar dockerfiles/run/guice/cassandra-rabbitmq/destination"
+                        sh "cp -r server/container/cli/target/james-server-cli.lib dockerfiles/run/guice/cassandra-rabbitmq/destination"
+                        sh "docker build -t james_run dockerfiles/run/guice/cassandra-rabbitmq"
+                    }
+                }
+            }
+        }
+        stage('Start James') {
+            steps {
+                script {
+                    docker.withTool('docker') {
+                        sh "docker run -d --name=cassandra cassandra:3.11.3"
```

Comment:
do you think that using docker compose here makes more sense?

Reply:
Yes, but how to tell which image of James to use?

I had a look on default Jenkinsfile steps, but there's none for docker-compose.
Should be great to contribute to it: ;)